File: Examples/csharp/lib/MyCdkAppStack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.SQS;
using Amazon.CDK.AWS.SNS;
using Amazon.CDK.AWS.SNS.Subscriptions;

namespace ApexCloudSuite
{
    public class ApexInfraStack : Stack
    {
        internal ApexInfraStack(Construct scope, string id, IStackProps? props = null)
            : base(scope, id, props)
        {
            var queue = new Queue(this, "ProcessingQueue", new QueueProps
            {
                VisibilityTimeout = Duration.Seconds(300)
            });

            var topic = new Topic(this, "NotificationTopic");

            topic.AddSubscription(new SqsSubscription(queue));
        }
    }
}

File: Examples/fsharp/lib/MyCdkAppStack.fs
namespace ApexCloudSuite

open Amazon.CDK
open Amazon.CDK.AWS.SQS
open Amazon.CDK.AWS.SNS
open Amazon.CDK.AWS.SNS.Subscriptions

type ApexInfraStack(scope: Construct, id: string, ?props: IStackProps) as this =
    inherit Stack(scope, id, defaultArg props null)

    let queue = Queue(this, "ProcessingQueue", QueueProps(VisibilityTimeout = Duration.Seconds(300.0)))

    let topic = Topic(this, "NotificationTopic")

    do topic.AddSubscription(SqsSubscription(queue)) |> ignore